src/CommonGLPI.php

```diff
@@ -1057,6 +1057,28 @@ class='btn btn-sm btn-icon btn-ghost-secondary'
    function display($options = []) {
       global $CFG_GLPI;
 
+      $redirect = false;
+
+      // Manage profile change
+      if (isset($_GET["change_profile"])) {
+         if (isset($_SESSION['glpiprofiles'][$_GET["change_profile"]])) {
+            Session::changeProfile($_GET["change_profile"]);
+         }
+         $redirect = true;
+      }
+
+      // Manage entity change
+      if (isset($_GET["change_entity"])) {
+         Session::changeActiveEntities($_GET["change_entity"], true);
+         $redirect = true;
+      }
+
+      if ($redirect) {
+         Html::redirect(
+            preg_replace("/(\?|&|".urlencode('?')."|".urlencode('&').")?(change_entity|change_profile).*/",
```

Comment:
1. If `change_entity` or `change_profile` is placed before some orther parameters, they will be removed.
2. I guess you used `HTTP_REFERER` to users that are not already logged-in, but for users that do not have to use login form, should this logic be applied to `REQUEST_URI` and `QUERY_STRING` keys ?